Allow scala version to be explicitly declared (#31118)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

Nested events are reflected in the HTML test reports, providing clear traceability.

#### Scala version can be declared explicitly

> Starting in this version of Gradle, when applying the `scala-base` or `scala` plugins, the version of Scala can now be declared explicitly on the `scala` extension.

Reviewer: ```suggestion
Starting in this version of Gradle, when applying the `scala-base` or `scala` plugins, you can now explicitly declare the Scala version on the `scala` extension.
```

> Based on the declared version, Gradle can derive the required Scala toolchain dependencies without requiring them to be declared by the user.

Reviewer: ```suggestion
This allows Gradle to automatically resolve the required Scala toolchain dependencies, eliminating the need for the user to declare them manually.
```

> Further, this allows Gradle to avoid inferring the Scala version from the production runtime classpath, a process which can be error-prone.

Reviewer: ```suggestion
It also removes the need to infer the Scala version from the production runtime classpath, which was error-prone.
```

> Previously, a `scala-library` dependency was required:

Reviewer: ```suggestion
Previously, you had to declare a `scala-library` dependency, like this:

```

```kotlin
plugins {
    id("scala")
}

repositories {
    mavenCentral()
}

dependencies {
    implementation("org.scala-lang:scala-library:2.13.12")
    // OR
    implementation("org.scala-lang:scala3-library_3:3.6.3")
}
```

> Now, the Scala version can be declared explicitly and the `scala-library` dependency is no longer required:

Reviewer: ```suggestion
Now, you can explicitly set the Scala version in the `scala` extension, and the `scala-library` dependency is no longer required:
```

Nested events are reflected in the HTML test reports, providing clear traceability.

#### Scala version can be declared explicitly

> Starting in this version of Gradle, when applying the `scala-base` or `scala` plugins, you can now explicitly declare the Scala version on the `scala` extension.

Reviewer 2: can this link to the scala plugin page?
